ni845x_spi.c

```diff
@@ -0,0 +1,501 @@
+                msg_pwarn("USB-8452 IO Voltage coerced to: %.1f V\n", (float)selected_voltage_100mV / 10.0f);
+			} else {
+                msg_pwarn("USB-8452 IO Voltage set to: %.1f V\n", (float)selected_voltage_100mV / 10.0f);
+			}
+            break;
+        }
+    }
+
+    if (selected_voltage_100mV == 0) {
+        msg_perr("The USB-8452 does not supports the %.1f V IO voltage\n", IOVoltage_mV/1000.0f);
+        msg_perr("Supported IO voltages: ");
+        for (i = 0; i < ARRAY_SIZE(usb8452_io_voltages_in_100mV); i++) {
+            msg_perr("%.1fV", (double)usb8452_io_voltages_in_100mV[i]/1000.0f);
+            if (i != ARRAY_SIZE(usb8452_io_voltages_in_100mV) - 1)
+                msg_perr(", ");
+        }
+        msg_perr("\n");
+        return -1;
+    }
+
+    i = ni845xSetIoVoltageLevel(device_handle, selected_voltage_100mV);
+    if (i != 0) {
+        ni845xStatusToString(i, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_perr("ni845xSetIoVoltageLevel failed with: %s (%d)\n", error_string_buffer, i);
+        return -1;
+    }
+    return 0;
+}
+
+/**
+ * @brief ni845x_spi_set_speed sets the SPI SCK speed
+ * @param clockToSetInkHz SCK speed in KHz
+ * @return
+ */
+static int ni845x_spi_set_speed(uint16_t clockToSetInkHz)
+{
+    int32 i = ni845xSpiConfigurationSetClockRate(configuration_handle, clockToSetInkHz);
+    uInt16 clock_freq_read_KHz;
+    if (i != 0) {
+        ni845xStatusToString(i, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_perr("ni845xSpiConfigurationSetClockRate(%d) failed with: %s (%ld)\n",
+                 clockToSetInkHz,
+                 error_string_buffer,
+                 i);
+        return -1;
+    }
+
+    // read back the clock frequency and notify the user if it is not the same as it was requested
+    i = ni845xSpiConfigurationGetClockRate(configuration_handle, &clock_freq_read_KHz);
+    if (i != 0) {
+        ni845xStatusToString(i, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_perr("ni845xSpiConfigurationSetClockRate failed with: %s (%ld)\n", error_string_buffer, i);
+        return -1;
+    } else {
+        if (clock_freq_read_KHz != clockToSetInkHz) {
+            msg_pinfo("SPI clock frequency coerced to: %d KHz (requested: %d KHz)\n",
+                      (int)clock_freq_read_KHz,
+                      (int)clockToSetInkHz);
+        } else {
+            msg_pinfo("SPI clock frequency set to: %d KHz\n",
+                      (int)clockToSetInkHz);
+        }
+        return 0;
+    }
+}
+
+/**
+ * @brief ni845x_spi_print_available_devices prints a list of the available devices
+ */
+void ni845x_spi_print_available_devices(void)
+{
+    char resource_handle[256], device_type_string[16];
+    NiHandle device_find_handle;
+    uInt32 found_devices_count = 0;
+    int32 tmp = 0;
+    uint32_t pid, vid, usb_bus, serial_as_number;
+
+    tmp = ni845xFindDevice(resource_handle, &device_find_handle, &found_devices_count);
```

Comment:
Check the value of tmp so we know the stuff below will work as expected.

Reply:
Makes sense, fixed.